python/ray/experimental/collective/allreduce.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from typing import List, Optional
+
+import ray
+from ray.dag.collective_node import CollectiveGroup, CollectiveOutputNode
+from ray.dag.constants import (
+    BIND_INDEX_KEY,
+    COLLECTIVE_GROUP_KEY,
+    PARENT_CLASS_NODE_KEY,
+)
+from ray.dag.dag_node import DAGNode
+from ray.util.collective import types
+
+logger = logging.getLogger(__name__)
+
+
+class AllReduceWrapper:
+    # [TODO] Comment.
+
+    def bind(
+        self,
+        input_nodes: List["DAGNode"],
+        op: types.ReduceOp,
+        count: Optional[int] = None,
```

Comment:
Let's remove this from the user-facing API.

Let's also set a default argument (SUM) for `op`.

Reply:
Done.